aws-ts-organizations/backupPolicy.ts

```diff
@@ -0,0 +1,202 @@
+    orgUnitId: pulumi.Input<string>;
+    /**
+     * A map of the accounts with the `key` being a short-name
+     * for the account and the `value` is the account resource.
+     */
+    accounts: { [key: string]: aws.organizations.Account };
+    assumeRoleName: string;
+}
+
+/**
+ * Creates a backup policy for the organizational unit (OU).
+ * This component also creates backup vaults in each of the
+ * target accounts that the backup policy requires.
+ */
+export class BackupPolicy extends pulumi.ComponentResource {
+    accounts: { [key: string]: aws.organizations.Account };
+    assumeRoleName: string;
+    /**
+     * Region where the backup should be copied to.
+     * This is also sometimes known as the secondary
+     * backup region. The primary is wherever the
+     * resource being backed up lives. AWS Backup
+     * calls this the "copy" region.
+     */
+    backupRegion: aws.Region;
+    orgUnitId: pulumi.Input<string>;
+    primaryRegions: aws.Region[];
+
+    monthlyBackupPolicyIamRoleName = "MyMonthlyBackupIamRole";
+
+    constructor(
+        name: string,
+        args: BackupPolicyArgs,
+        opts?: pulumi.ComponentResourceOptions
+    ) {
+        super("acme:policies:BackupPolicy", name, undefined, opts);
+
+        this.accounts = args.accounts;
+        this.assumeRoleName = args.assumeRoleName;
+        this.backupRegion = args.backupRegion;
+        this.orgUnitId = args.orgUnitId;
+        this.primaryRegions = args.primaryRegions;
+
+        for (const accountName in args.accounts) {
+            const account = args.accounts[accountName];
+            const accountProvider = new aws.Provider(
+                `${accountName}Provider`,
+                {
+                    assumeRole: {
+                        roleArn: pulumi.interpolate`arn:aws:iam::${account.id}:role/${this.assumeRoleName}`,
+                    },
+                    allowedAccountIds: [account.id],
+                },
+                { parent: this }
+            );
+
+            this.createBackupVault(accountName, accountProvider);
+
+            this.createBackupPolicyIamRole(accountName, accountProvider);
+        }
+
+        this.createMonthlyBackupPolicy();
+
+        super.registerOutputs();
+    }
+
+    /**
+     * Creates a backup vault in the copy region.
+     */
+    private createBackupVault(accountName: string, accountProvider: Provider) {
+        const vault = new aws.backup.Vault(
+            `backupVault-${accountName}`,
+            // We rely on the vault name staying consistent across AWS accounts
+            // for the purposes of the backup policy that is attached to the
+            // organizational unit.
+            { name: "Default" },
+            { provider: accountProvider, parent: this }
+        );
+    }
+
+    private createBackupPolicyIamRole(
+        accountName: string,
+        accountProvider: Provider
+    ) {
+        const backupPolicyRole = new aws.iam.Role(
```

Comment:
When I ran this with my personal account (which has an organization and several child accounts), I got this error:

```
Diagnostics:
  aws:iam:Role (devAccountBackupPolicyRole):
    error: 1 error occurred:
        * configuring Terraform AWS Provider: error validating provider credentials: error calling sts:GetCallerIdentity: operation error STS: GetCallerIdentity, failed to sign request: failed to retrieve credentials: failed to refresh cached credentials, operation error STS: AssumeRole, https response error StatusCode: 403, RequestID: 2524af3f-a926-4be8-886f-c1c63a21c167, RegionDisabledException: STS is not activated in this region for account:334166260394. Your account administrator can activate STS in this region using the IAM Console.
```

To get this to work, I had to navigate to IAM Identity Center and click Enable -- although to be honest I'm not entirely sure why that had to be done. We should explain that this needs to be done as well, and why.

Reply:
I've add this to the README.